contracts/leaser/src/migrate.rs

```diff
@@ -2,101 +2,305 @@ use lease::api::MigrateMsg;
+            exp.next_customer = None;
+            assert_eq!(
+                Ok(exp),
+                super::migrate_leases(test_customers(), new_code, 7)
+            );
+        }
     }
 
     #[test]
     fn err_leases() {
         let new_code = 242;
+        let lease1 = Addr::unchecked("lease11");
+        let lease2 = Addr::unchecked("lease12");
+        let lease3 = Addr::unchecked("lease13");
+        let cust1 = Customer::from(
+            Addr::unchecked("customer1"),
+            vec![lease1, lease2, lease3].into_iter(),
+        );
         let err = "testing error";
 
-        let no_leases = vec![
-            Ok(Addr::unchecked("fsdffg")),
+        let customers = vec![
+            Ok(cust1),
             Err(ContractError::ParseError { err: err.into() }),
-            Ok(Addr::unchecked("2424")),
         ];
         assert_eq!(
             Err(ContractError::ParseError { err: err.into() }),
-            super::migrate_leases(no_leases.into_iter(), new_code)
+            super::migrate_leases(customers.into_iter(), new_code, 3)
         );
     }
 
-    #[test]
-    fn a_few_leases() {
-        let new_code = 242;
-        let addr1 = Addr::unchecked("11111");
-        let addr2 = Addr::unchecked("22222");
-        let leases = vec![addr1.clone(), addr2.clone()];
-
-        let mut exp = Batch::default();
-        exp.schedule_migrate_wasm_no_reply(&addr1, MigrateMsg {}, new_code)
-            .unwrap();
-        exp.schedule_migrate_wasm_no_reply(&addr2, MigrateMsg {}, new_code)
+    fn add_expected(mut exp: MigrationResult, lease_addr: &Addr, new_code: u64) -> MigrationResult {
+        exp.msgs
+            .schedule_migrate_wasm_no_reply(lease_addr, MigrateMsg {}, new_code)
             .unwrap();
+        exp
+    }
 
-        assert_eq!(
-            Ok(exp),
-            super::migrate_leases(leases.into_iter().map(Ok), new_code)
-        );
+    fn test_customers() -> impl Iterator<Item = ContractResult<Customer<IntoIter<Addr>>>> {
+        let lease1 = Addr::unchecked(LEASE1);
+        let lease21 = Addr::unchecked(LEASE21);
+        let lease22 = Addr::unchecked(LEASE22);
+        let lease3 = Addr::unchecked(LEASE3);
+        let lease41 = Addr::unchecked(LEASE41);
+        let lease42 = Addr::unchecked(LEASE42);
+        let lease43 = Addr::unchecked(LEASE43);
+        let customer_addr1 = Addr::unchecked(CUSTOMER_ADDR1);
+        let customer_addr2 = Addr::unchecked(CUSTOMER_ADDR2);
+        let customer_addr3 = Addr::unchecked(CUSTOMER_ADDR3);
+        let customer_addr4 = Addr::unchecked(CUSTOMER_ADDR4);
+        let cust1 = Customer::from(customer_addr1, vec![lease1].into_iter());
+        let cust2 = Customer::from(customer_addr2, vec![lease21, lease22].into_iter());
+        let cust3 = Customer::from(customer_addr3, vec![lease3].into_iter());
+        let cust4 = Customer::from(customer_addr4, vec![lease41, lease42, lease43].into_iter());
```

Comment:
I'd suggest at least grouping those for better readability.
```suggestion
        let lease1 = Addr::unchecked(LEASE1);
        let customer_addr1 = Addr::unchecked(CUSTOMER_ADDR1);
        let cust1 = Customer::from(customer_addr1, vec![lease1].into_iter());

        let lease21 = Addr::unchecked(LEASE21);
        let lease22 = Addr::unchecked(LEASE22);
        let customer_addr2 = Addr::unchecked(CUSTOMER_ADDR2);
        let cust2 = Customer::from(customer_addr2, vec![lease21, lease22].into_iter());

        let lease3 = Addr::unchecked(LEASE3);
        let customer_addr3 = Addr::unchecked(CUSTOMER_ADDR3);
        let cust3 = Customer::from(customer_addr3, vec![lease3].into_iter());

        let lease41 = Addr::unchecked(LEASE41);
        let lease42 = Addr::unchecked(LEASE42);
        let lease43 = Addr::unchecked(LEASE43);
        let customer_addr4 = Addr::unchecked(CUSTOMER_ADDR4);
        let cust4 = Customer::from(customer_addr4, vec![lease41, lease42, lease43].into_iter());
```